### ISSUE TITLE
[python-pillow] DLL load failed while importing _imaging: The specified module could not be found

### ISSUE THREAD

**Issue by augusto-rehfeldt** [msys2/MINGW-packages#11103]
Good evening.

```
C:\Users\Augusto\Desktop\python\pruebas_math>python3 prueba.py
Traceback (most recent call last):
  File "C:\Users\Augusto\Desktop\python\pruebas_math\prueba.py", line 2, in <module>
    from PIL import Image, ImageOps, ImageFont, ImageDraw
  File "C:\msys64\mingw64\lib\python3.9\site-packages\PIL\Image.py", line 89, in <module>
    from . import _imaging as core
ImportError: DLL load failed while importing _imaging: The specified module could not be found.
```

```
C:\Users\Augusto\Desktop\python\pruebas_math>python3 -m pip install --upgrade Pillow
Requirement already satisfied: Pillow in c:\msys64\mingw64\lib\python3.9\site-packages (9.0.1)
```

I've followed [https://pillow.readthedocs.io/en/latest/installation.html#building-on-windows-using-msys2-mingw](https://pillow.readthedocs.io/en/latest/installation.html#building-on-windows-using-msys2-mingw) and installed all necessary dependencies.

I have Python 3.9.6.

I don't know what else to do. Nothing I find on the Internet works. I cannot even download it with pip as it gives me this error, only can build it with `pacman -S mingw-w64-x86_64-python-pillow`:

<details><summary>Click here to see the log</summary>

```
C:\Users\Augusto\Desktop\python\pruebas_math>C:\msys64\mingw64\bin\pip.exe install Pillow
Collecting Pillow
  Using cached Pillow-9.0.1.tar.gz (49.5 MB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: Pillow
  Building wheel for Pillow (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [160 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.mingw_x86_64-3.9
      creating build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BdfFontFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BlpImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BmpImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BufrStubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ContainerIO.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\CurImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\DcxImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\DdsImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\EpsImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ExifTags.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\features.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FitsStubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FliImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FontFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FpxImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FtexImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GbrImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GdImageFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GifImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GimpGradientFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GimpPaletteFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GribStubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\Hdf5StubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\IcnsImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\IcoImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\Image.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageChops.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageCms.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageColor.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageDraw.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageDraw2.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageEnhance.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageFilter.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageFont.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageGrab.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageMath.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageMode.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageMorph.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageOps.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImagePalette.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImagePath.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageQt.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageSequence.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageShow.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageStat.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageTk.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageTransform.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageWin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImtImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\IptcImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\Jpeg2KImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\JpegImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\JpegPresets.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\McIdasImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MicImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MpegImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MpoImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MspImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PaletteFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PalmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PcdImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PcfFontFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PcxImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PdfImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PdfParser.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PixarImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PngImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PpmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PsdImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PSDraw.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PyAccess.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\SgiImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\SpiderImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\SunImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TarIO.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TgaImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TiffImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TiffTags.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\WalImageFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\WebPImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\WmfImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\XbmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\XpmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\XVThumbImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_binary.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_tkinter_finder.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_util.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_version.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\__init__.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\__main__.py -> build\lib.mingw_x86_64-3.9\PIL
      running egg_info
      writing src\Pillow.egg-info\PKG-INFO
      writing dependency_links to src\Pillow.egg-info\dependency_links.txt
      writing top-level names to src\Pillow.egg-info\top_level.txt
      reading manifest file 'src\Pillow.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.c'
      warning: no files found matching '*.h'
      warning: no files found matching '*.sh'
      warning: no previously-included files found matching '.appveyor.yml'
      warning: no previously-included files found matching '.clang-format'
      warning: no previously-included files found matching '.coveragerc'
      warning: no previously-included files found matching '.editorconfig'
      warning: no previously-included files found matching '.readthedocs.yml'
      warning: no previously-included files found matching 'codecov.yml'
      warning: no previously-included files matching '.git*' found anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      no previously-included directories found matching '.ci'
      adding license file 'LICENSE'
      writing manifest file 'src\Pillow.egg-info\SOURCES.txt'
      running build_ext
      dllwrap: WARNING: dllwrap is deprecated, use gcc -shared or ld -shared instead

      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\Augusto\AppData\Local\Temp\pip-install-ol0emoel\pillow_0f6531e1a2314c03ba03f03ad76194a0\setup.py", line 989, in <module>
          setup(
        File "C:\msys64\mingw64\lib\python3.9\site-packages\setuptools\__init__.py", line 153, in setup
          return distutils.core.setup(**attrs)
        File "C:\msys64\mingw64\lib\python3.9\distutils\core.py", line 148, in setup
          dist.run_commands()
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 966, in run_commands
          self.run_command(cmd)
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 985, in run_command
          cmd_obj.run()
        File "C:\msys64\mingw64\lib\python3.9\site-packages\wheel\bdist_wheel.py", line 299, in run
          self.run_command('build')
        File "C:\msys64\mingw64\lib\python3.9\distutils\cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 985, in run_command
          cmd_obj.run()
        File "C:\msys64\mingw64\lib\python3.9\distutils\command\build.py", line 135, in run
          self.run_command(cmd_name)
        File "C:\msys64\mingw64\lib\python3.9\distutils\cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 985, in run_command
          cmd_obj.run()
        File "C:\msys64\mingw64\lib\python3.9\site-packages\setuptools\command\build_ext.py", line 79, in run
          _build_ext.run(self)
        File "C:\msys64\mingw64\lib\python3.9\site-packages\Cython\Distutils\old_build_ext.py", line 186, in run
          _build_ext.build_ext.run(self)
        File "C:\msys64\mingw64\lib\python3.9\distutils\command\build_ext.py", line 307, in run
          self.compiler = new_compiler(compiler=self.compiler,
        File "C:\msys64\mingw64\lib\python3.9\distutils\ccompiler.py", line 1034, in new_compiler
          return klass(None, dry_run, force)
        File "C:\msys64\mingw64\lib\python3.9\distutils\cygwinccompiler.py", line 308, in __init__
          CygwinCCompiler.__init__ (self, verbose, dry_run, force)
        File "C:\msys64\mingw64\lib\python3.9\distutils\cygwinccompiler.py", line 125, in __init__
          if self.ld_version >= "2.10.90":
      TypeError: '>=' not supported between instances of 'NoneType' and 'str'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for Pillow
  Running setup.py clean for Pillow
Failed to build Pillow
Installing collected packages: Pillow
  Running setup.py install for Pillow ... error
  error: subprocess-exited-with-error

  × Running setup.py install for Pillow did not run successfully.
  │ exit code: 1
  ╰─> [162 lines of output]
      running install
      running build
      running build_py
      creating build
      creating build\lib.mingw_x86_64-3.9
      creating build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BdfFontFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BlpImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BmpImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\BufrStubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ContainerIO.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\CurImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\DcxImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\DdsImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\EpsImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ExifTags.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\features.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FitsStubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FliImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FontFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FpxImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\FtexImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GbrImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GdImageFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GifImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GimpGradientFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GimpPaletteFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\GribStubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\Hdf5StubImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\IcnsImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\IcoImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\Image.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageChops.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageCms.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageColor.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageDraw.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageDraw2.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageEnhance.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageFilter.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageFont.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageGrab.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageMath.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageMode.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageMorph.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageOps.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImagePalette.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImagePath.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageQt.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageSequence.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageShow.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageStat.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageTk.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageTransform.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImageWin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\ImtImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\IptcImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\Jpeg2KImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\JpegImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\JpegPresets.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\McIdasImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MicImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MpegImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MpoImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\MspImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PaletteFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PalmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PcdImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PcfFontFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PcxImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PdfImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PdfParser.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PixarImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PngImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PpmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PsdImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PSDraw.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\PyAccess.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\SgiImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\SpiderImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\SunImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TarIO.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TgaImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TiffImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\TiffTags.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\WalImageFile.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\WebPImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\WmfImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\XbmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\XpmImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\XVThumbImagePlugin.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_binary.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_tkinter_finder.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_util.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\_version.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\__init__.py -> build\lib.mingw_x86_64-3.9\PIL
      copying src\PIL\__main__.py -> build\lib.mingw_x86_64-3.9\PIL
      running egg_info
      writing src\Pillow.egg-info\PKG-INFO
      writing dependency_links to src\Pillow.egg-info\dependency_links.txt
      writing top-level names to src\Pillow.egg-info\top_level.txt
      reading manifest file 'src\Pillow.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.c'
      warning: no files found matching '*.h'
      warning: no files found matching '*.sh'
      warning: no previously-included files found matching '.appveyor.yml'
      warning: no previously-included files found matching '.clang-format'
      warning: no previously-included files found matching '.coveragerc'
      warning: no previously-included files found matching '.editorconfig'
      warning: no previously-included files found matching '.readthedocs.yml'
      warning: no previously-included files found matching 'codecov.yml'
      warning: no previously-included files matching '.git*' found anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      no previously-included directories found matching '.ci'
      adding license file 'LICENSE'
      writing manifest file 'src\Pillow.egg-info\SOURCES.txt'
      running build_ext
      dllwrap: WARNING: dllwrap is deprecated, use gcc -shared or ld -shared instead

      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "C:\Users\Augusto\AppData\Local\Temp\pip-install-ol0emoel\pillow_0f6531e1a2314c03ba03f03ad76194a0\setup.py", line 989, in <module>
          setup(
        File "C:\msys64\mingw64\lib\python3.9\site-packages\setuptools\__init__.py", line 153, in setup
          return distutils.core.setup(**attrs)
        File "C:\msys64\mingw64\lib\python3.9\distutils\core.py", line 148, in setup
          dist.run_commands()
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 966, in run_commands
          self.run_command(cmd)
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 985, in run_command
          cmd_obj.run()
        File "C:\msys64\mingw64\lib\python3.9\site-packages\setuptools\command\install.py", line 61, in run
          return orig.install.run(self)
        File "C:\msys64\mingw64\lib\python3.9\distutils\command\install.py", line 547, in run
          self.run_command('build')
        File "C:\msys64\mingw64\lib\python3.9\distutils\cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 985, in run_command
          cmd_obj.run()
        File "C:\msys64\mingw64\lib\python3.9\distutils\command\build.py", line 135, in run
          self.run_command(cmd_name)
        File "C:\msys64\mingw64\lib\python3.9\distutils\cmd.py", line 313, in run_command
          self.distribution.run_command(command)
        File "C:\msys64\mingw64\lib\python3.9\distutils\dist.py", line 985, in run_command
          cmd_obj.run()
        File "C:\msys64\mingw64\lib\python3.9\site-packages\setuptools\command\build_ext.py", line 79, in run
          _build_ext.run(self)
        File "C:\msys64\mingw64\lib\python3.9\site-packages\Cython\Distutils\old_build_ext.py", line 186, in run
          _build_ext.build_ext.run(self)
        File "C:\msys64\mingw64\lib\python3.9\distutils\command\build_ext.py", line 307, in run
          self.compiler = new_compiler(compiler=self.compiler,
        File "C:\msys64\mingw64\lib\python3.9\distutils\ccompiler.py", line 1034, in new_compiler
          return klass(None, dry_run, force)
        File "C:\msys64\mingw64\lib\python3.9\distutils\cygwinccompiler.py", line 308, in __init__
          CygwinCCompiler.__init__ (self, verbose, dry_run, force)
        File "C:\msys64\mingw64\lib\python3.9\distutils\cygwinccompiler.py", line 125, in __init__
          if self.ld_version >= "2.10.90":
      TypeError: '>=' not supported between instances of 'NoneType' and 'str'
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> Pillow

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

</details>

Thanks for helping.

**Comment by Biswa96** [comment]
Both commands works in my system.

```
python -c "from PIL import _imaging"
python -c "from PIL import Image, ImageOps, ImageFont, ImageDraw"
```

**Comment by augusto-rehfeldt** [comment]
> Both commands works in my system.
> 
> ```
> python -c "from PIL import _imaging"
> python -c "from PIL import Image, ImageOps, ImageFont, ImageDraw"
> ```

But why? Could there be a conflict? What can I do? What Pillow version do you have installed on Msys? Thanks a lot for the quick reply.

**Comment by Biswa96** [comment]
Some guess.

1. Make sure there is no other python installation which can conflicts with mingw python.
2. From the first output, I guess you are using Command Prompt to run mingw python. Try with mingw environment.
3. Use `pacman -S $MINGW_PACKAGE_PREFIX-python-pillow` command to install the package instead of pip.

**Comment by augusto-rehfeldt** [comment]
Good morning.

1 - I uninstalled all other installations to no use.
2 - I did, nothing happened.
3 - This is what I had to do originally as pip gave me the `#error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."` error.

I don't know what else could it be!

**Comment by bsam-parsionate** [comment]
Downgrading the pillow package to 9.0.0 seems to be working for me. `pip install pillow==9.0.0`

**Comment by KlepD-SAL** [comment]
The error still persists (for me, at least).

> Downgrading the `pillow` package to `9.0.0` seems to be working for me.
> 
> `pip install pillow==9.0.0`

Unfortunately, this doesn't work for me on MSYS.

I tried to use PIP, but it always threw errors while building (even for numpy etc ..).
Therefore I wanted to stick with the MSYS packages.
Can one safely downgrade python packages there?

**Comment by Biswa96** [comment]
Packages from pip may not be compatible with mingw python, see https://www.msys2.org/docs/python/

Please provide steps to reproduce the issue after installing `$MINGW_PACKAGE_PREFIX-python-pillow` package.

**Comment by CBanafo** [comment]
@bsam-parsionate 
You truly saved my day man. God bless....

**Comment by Omahqwen** [comment]
@bsam-parsionate ,this worked for me, thanks so much.

**Comment by eliandata** [comment]
> Downgrading the `pillow` package to `9.0.0` seems to be working for me.
> 
> `pip install pillow==9.0.0`

**Comment by eliandata** [comment]
This works for me!

**Comment by irwanrei** [comment]
> Downgrading the `pillow` package to `9.0.0` seems to be working for me.
> 
> `pip install pillow==9.0.0`

It works. Thank you.

**Comment by augusto-rehfeldt** [comment]
> Downgrading the `pillow` package to `9.0.0` seems to be working for me.
> 
> `pip install pillow==9.0.0`

I never replied to this lol.

This did the trick in that case. Thanks.

**Comment by Nik-Kras** [comment]
Downgrading python to 3.6 helped me

**Comment by aleenarayamajhi** [comment]
> pip install pillow==9.0.0

Thanks! It worked for me as well!

**Comment by gobears21** [comment]
> Downgrading the `pillow` package to `9.0.0` seems to be working for me.
> 
> `pip install pillow==9.0.0`

this method works fine for me

**Comment by austinjoseph** [comment]
Having a hard time getting Conda to downgrade to pillow==9.0.0. Anyone else solve this problem?

**Comment by gobears21** [comment]
> Having a hard time getting Conda to downgrade to pillow==9.0.0. Anyone else solve this problem?

trying using `pip install pillow==9.0.0`?
if you are in the conda env, all packages installed through pip will be stored in the conda directory.

**Comment by austinjoseph** [comment]
Sorry I forgot to follow up with:

To workaround this I forced conda to run python 3.10 (Pillow 9.0.* is not supported for python 3.11+)
Then just installed pillow=9.0.1

It's working in my env now

**Comment by mfatih7** [comment]
I get this Import Error while trying to build an environment for pytorch 2.0.1 and cuda 11.8.

If the environment is built on Python 3.11 and Pytorch is installed first with

`conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia`

The Import Error occurs. To solve the problem I generated a new environment on Python 3.11

First installed the latest version of PIL(pillow==10.0.0) using

`conda install pillow`

Then I installed Pytorch 2.0.1 using

`conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia`

The import error does not occur.
